Update README; add .mailmap

cxw42 opened the pull request:
A few minor things.  We later need to finish updates per https://github.com/editorconfig/specification/pull/17#issue-400585616 .  However, I suggest we not do that until after the vote forthcoming per https://github.com/editorconfig/editorconfig/issues/395#issuecomment-628812024 .  Therefore, those changes are not in this PR.

In the mean time:
- Update README: fix some typos, and mention the versioning tie-in with
  the specification after editorconfig/editorconfig-vote#11.
- Add .mailmap: this way the automatically-generated Git lists of
  authors and committers only have one entry per person.

greut commented on README.md:
```diff
@@ -29,9 +31,15 @@ set(EDITORCONFIG_CMD the_editorconfig_core_path)
+[specification][].
 
 [cmake]: http://www.cmake.org
-[editorconfig]: http://editorconfig.org
```
https://editorconfig.org

cxw42 replied:
Fixed, and the other URLs.  Thanks!

amanda4581 reviewed (dismissed):
>

cxw42 committed:
Update README; add .mailmap

- Update README: fix some typos, and mention the versioning tie-in with
  the specification after editorconfig/editorconfig-vote#11.
- Add .mailmap: this way the automatically-generated Git lists of
  authors and committers only have one entry per person.
